docs/proposals/operator-managed-infra.md

```diff
@@ -2,73 +2,56 @@
+* Enterprises generally have very specific requirements about how they create their K8s Clusters and nodepools.
+* Allow Substratus to work in a multi namespace environment. Currently namespace is hardcoded in workload identity settings in the terraform
 
 ## How?
-* Make the Image Registry and Object Storage Bucket configurable as a setting of the operator itself
-* Object storage will be optional and if not provided then default PVC class will be used
-* Utilize a single service account that has enough permissions to do everything needed within Substratus
-* Simplify Substratus to use a single K8s SA per namespace. Simplicity is preferred here over the minor security benefits you get
-  for using a different SA for each kind of Substratus resource.
-* Provide Substratus controller the permissions to manage registry, bucket and set IAM policy on the GCP SA.
-  The following permissions will be required by the substratus Service Account on the project itself:
-  ```
-  roles/storage.admin, roles/artifactregistry.repoAdmin, roles/iam.serviceAccountAdmin
-  ```
 
-### Implementation details
-* Object storage: Create bucket if the configured bucket does not exist
-* Image Registry: Create registry if the configured registry does not exist
-* New sub reconciler for namespace: Create K8s Service Accounts automatically in each namespace where Substratus is used and
-  annotate the K8s Service Account with the correct GCP Service Account. Ensure that following
-  gets called when a new Service Account is provisioned in a new namespace:
+<img src="./diagrams/operator-managed-infra.md" width="80%"></img>
+
+### Installation
+**Installation Flow** will be responsible for the following:
+* Create a K8s cluster with nodes/GPUs, this step is optional if end-user already has a K8s cluster (eksctl or gcloud)
+* Create Bucket, this step is optional if end-user already has a bucket
+* Create Image Registry, this step is optional if end-user already has an image registry
+* REQUIRED: Create Google Service Account or AWS IAM Role that has resource level permissions to:
+  * read and write a specific bucket
+  * pull and push to a specific Image Registry (GAR/ECR)
+  * (GCP) set IAM policy on the service Account itself to be able to manage workload identity bindings to KSAs. This can be done by assigning the service account to be a ServiceAccountAdmin to itself
+  * (AWS) UpdateAssumeRolePolicy on the AWS Role used by Substratus (required for multi namespace support)
+* REQUIRED: Apply Subsubstratus manifests and configure them to use correct bucket and image registry
+
+### Controller
+Controller will be responsible for the following:
+* Creating the K8s ServiceAccount and related binding by calling `enforceServiceAccount` whenever a Substratus resource gets reconciled
+* (GCP only) example set annotation for Google Service Account AND update IAM policy ton the Service Account so it can use the Google Service Account. For example:
```

Comment:
```suggestion
* (GCP only) example set annotation for Google Service Account AND update IAM policy on the Service Account so it can use the Google Service Account. For example:
```